cmd/celestia-appd/cmd/root.go

```diff
@@ -92,63 +92,77 @@ func NewRootCmd() *cobra.Command {
+		panic(fmt.Errorf("failed to enhance root command: %w", err))
+	}
 
 	return rootCommand
 }
 
 // initRootCommand performs a bunch of side-effects on the root command.
-func initRootCommand(rootCommand *cobra.Command, encodingConfig encoding.Config) {
+func initRootCommand(rootCommand *cobra.Command, capp *app.App) {
 	debugCmd := debug.Cmd()
 	debugCmd.AddCommand(
+		NewInPlaceTestnetCmd(),
+		AppGenesisToCometGenesisConverterCmd(),
 		server.ModuleHashByHeightQuery(NewAppServer),
 	)
 
 	rootCommand.AddCommand(
-		genutilcli.InitCmd(app.ModuleBasics, app.DefaultNodeHome),
-		genutilcli.CollectGenTxsCmd(banktypes.GenesisBalancesIterator{}, app.DefaultNodeHome),
-		genutilcli.MigrateGenesisCmd(),
-		simdcmd.AddGenesisAccountCmd(app.DefaultNodeHome),
-		genutilcli.GenTxCmd(app.ModuleBasics, encodingConfig.TxConfig, banktypes.GenesisBalancesIterator{}, app.DefaultNodeHome),
-		genutilcli.ValidateGenesisCmd(app.ModuleBasics),
+		genutilcli.InitCmd(capp.BasicManager, app.DefaultNodeHome),
+		genutilcli.Commands(capp.GetTxConfig(), capp.BasicManager, app.DefaultNodeHome),
 		tmcli.NewCompletionCmd(rootCommand, true),
 		debugCmd,
-		clientconfig.Cmd(),
+		confixcmd.ConfigCommand(),
 		commands.CompactGoLevelDBCmd,
 		addrbookCommand(),
 		downloadGenesisCommand(),
 		addrConversionCmd(),
-		rpc.StatusCommand(),
-		queryCommand(),
-		txCommand(),
-		keys.Commands(app.DefaultNodeHome),
-		blobstreamclient.VerifyCmd(),
+		server.StatusCommand(),
+		queryCommand(capp.BasicManager),
+		txCommand(capp.BasicManager),
+		keys.Commands(),
 		snapshot.Cmd(NewAppServer),
 	)
 
-	// Add the following commands to the rootCommand: start, tendermint, export, version, and rollback.
-	addCommands(rootCommand, app.DefaultNodeHome, NewAppServer, appExporter, addStartFlags)
-}
+	modifyRootCommand(rootCommand)
 
-// setDefaultConsensusParams sets the default consensus parameters for the
-// embedded server context.
-func setDefaultConsensusParams(command *cobra.Command) error {
-	ctx := server.GetServerContextFromCmd(command)
-	ctx.DefaultConsensusParams = app.DefaultConsensusParams()
-	return server.SetCmdServerContext(command, ctx)
+	// find start command
+	startCmd, _, err := rootCommand.Find([]string{"start"})
+	if err != nil {
+		panic(fmt.Errorf("failed to find start command: %w", err))
+	}
+	startCmdRunE := startCmd.RunE
+
+	// Add the BBR check to the start command
+	startCmd.RunE = func(cmd *cobra.Command, args []string) error {
+		if err := checkBBR(cmd); err != nil {
+			return err
+		}
+
+		return startCmdRunE(cmd, args)
+	}
 }
 
 // addStartFlags adds flags to the start command.
 func addStartFlags(startCmd *cobra.Command) {
-	crisis.AddModuleInitFlags(startCmd)
 	startCmd.Flags().Int64(UpgradeHeightFlag, 0, "Upgrade height to switch from v1 to v2. Must be coordinated amongst all validators")
+	if err := startCmd.Flags().MarkDeprecated(UpgradeHeightFlag, "This flag is deprecated and was only useful prior to v4."); err != nil {
```

Comment:
[question] what if someone wants to use the v4 multiplexer binary to sync from genesis? Seems fine to deprecate on the v4 binary if we automatically start the embedded v3 binary with the apprpriate `--v2-upgrade-heights` which varies per chain-id. The v2 upgrade heights per chain-id are listed on https://docs.celestia.org/how-to-guides/network-upgrade-process#past-upgrades

Reply:
We can also bake the v2 upgrade heights directly into the binary instead of passing it via CLI or ldflags because the v2 heights won't change.

Reply:
The problem is that it isn't not handy for testing, and chain-id specific. Baking it in goreleaser depending on the tag suffix may be nicer.